bdk_chain/src/keychain.rs

```diff
@@ -221,6 +221,8 @@ pub struct Balance {
     pub untrusted_pending: u64,
     /// Confirmed and immediately spendable balance
     pub confirmed: u64,
+    /// UTXOs reserved by transactions not yet broadcasted
+    pub selected: u64,
```

Comment:
Instead of introducing a new field, another idea would be to have a `bool` input when calculating the `Balance` structure that filters out `reserved`:

```rust
    pub fn balance(&self, include_reserved: bool, mut should_trust: impl FnMut(&K) -> bool) -> Balance { todo!() }
```

Personally I would find this more useful, as it is rare to want the exact balance of the reserved UTXOs.

Reply:
I think it would also make sense to filter out `reserved` UTXOs from `KeychainTracker::full_utxos(&self, include_reserved: bool) -> ...`.

Reply:
I'm up for using `reserved`. Let's see what others think